src/conq/data_recorder.py

```diff
@@ -93,31 +93,42 @@ def save_episode_worker(self, clients, done: Event, mode: str, save_interval):
 
         while not done.is_set():
             now = time.time()
-            state = clients.state.get_robot_state()
-            localization_state = clients.graphnav.get_localization_state()
-
-            from conq.rerun_utils import viz_common_frames
-            viz_common_frames(state.kinematic_state.transforms_snapshot)
+            state = None
+            localization_state = None
 
+            # Initialize step_data
             step_data = {
                 'time': now,
-                'robot_state': state,
-                'localization': localization_state.localization,
-                'is_lost': localization_state.lost_detector_state.is_lost,
+                'robot_state': None,
```

Comment:
I would just not put them in a dict unless they're available. Seems better than storing a bunch of `None`s

Reply:
Agreed.

Reply:
This also saves you the need to initialize both the `state` and `localization_state` variables as `None`. Seems like you could remove those two lines?

Reply:
Good call! Fixed in [b0e352f](https://github.com/UM-ARM-Lab/conq_python/pull/18/commits/b0e352f9a508cd134fd9a16ae30fbc444118b45d)